### PR TITLE
Validation specs

### PR DESCRIPTION
**Pull request by imubarak234**
- Add the following validations:
   - For the User model:
     - Name must not be blank.
     - PostsCounter must be an integer greater than or equal to zero.
   - For the Post model:
     - Title must not be blank.
     - Title must not exceed 250 characters.
     - CommentsCounter must be an integer greater than or equal to zero.
     - LikesCounter must be an integer greater than or equal to zero.
 - Add unit specs for all of your models' methods and validations.
 - Add flash messages in the create actions in all your controllers.
 - Make sure that the N+1 problem is solved when fetching all posts and their comments for a user. This is the corresponding wireframe

### CODE REVIEW

File: app/controllers/likes_controller.rb [hunk 1]
```diff
@@ -4,6 +4,7 @@ def create
     puts @like
     flash[:notice] = @like.errors.full_messages.to_sentence unless @like.save
 
+    flash[:notice] = 'You have successfully liked this post.'
     redirect_to "http://127.0.0.1:3000/users/#{current_user.id}/posts/#{@like.post.id}"
```
Comment (MrBrN197) [inline app/controllers/likes_controller.rb]: - [ ] [OPTIONAL] you do not need to add the domain you are redirecting to. you simply need to use the path you want to go to.
- [ ] [OPTIONAL] you can also use rails' URL helpers to make this redirect more succinct. 

```suggestion
    flash[:notice] = 'You have successfully liked this post.'
    redirect_to user_post_path(current_user, post)
```

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -2,24 +2,26 @@ class PostsController < ApplicationController
   include ActionView::RecordIdentifier
 
   def index
-    @posts = Post.all
     @users = User.find_by(id: params[:user_id])
-    puts @posts
-  end
+    @users = User.new if @users.nil?
+    @posts = @users.posts
```
Comment (MrBrN197) [inline app/controllers/posts_controller.rb:7]: - [ ] Please address the N+1 problem when requesting all the posts for a user. ensure that you eagerly load the comments of the post when querying for the posts.

**Review by MrBrN197 — CHANGES_REQUESTED** [review]
## Hi, @imubarak234  👋 

You've done a really nice job so far.
There are some issues that you still need to work on to go to the next project but you are almost there!
## To highlight 👏👏
- ✔ All validation test cases are implemented and passing.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -21,7 +21,7 @@
       </div>
   </div>
 
-  <% @posts.each do |n| %>
+  <% @posts.includes(:comments).each do |n| %>
```
Comment (imubarak234) [inline app/views/posts/index.html.erb:24]: Thanks for the review. I used the 'include' for eager loading to also load the comments when I rendered the posts in views/posts/index.html.erb line 24

**Review by addod19 — APPROVED** [review]
### Hi @imubarak234, 

Your project is complete!  Nice job implementing the N+1 query problem :1st_place_medal: Awesome
There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**